src/components/Comment/Comment.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useMemo } from 'react';
+import './Comment.css';
+import InlineDotedList from 'Components/InlineDotedList';
+import cn from 'classnames';
+import User from 'Components/User';
+import LinkButton from 'Components/LinkButton';
+import CommentList from 'Components/CommentList';
+import CreateComment from 'Components/CreateComment';
+import ToggleButton from 'Components/ToggleButton';
+import type { Props } from './types';
+
+const Comment = ({ comment }: Props) => {
+    const [isCommentsOpen, setIsCommentsOpen] = useState(false);
+    const [isAnswerOpen, setIsAnswerOpen] = useState(false);
+
+    const isAnswer = useMemo(() => comment.answers === undefined, [comment.answers]);
+
+    const hasComments = useMemo(() => (
+        comment.answers ? Boolean(comment.answers.length) : false
+    ), [comment.answers]);
```

Comment:
1. Название `hasComments` немного расходится с названием сущности  `comment.answers`. Я бы назвал `hasAnswers`
2. Можно использовать `optional chaining` оператор

```suggestion
    const hasAnswers = useMemo(() => Boolean(comment.answers?.length), [comment.answers]);
```